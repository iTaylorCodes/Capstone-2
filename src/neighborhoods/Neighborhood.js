import { useSelector } from "react-redux";
import { useParams, useNavigate } from "react-router-dom";
import Scores from "./Scores";

/** Neighborhood.
 *
 * Component dedicated to the neighborhood being searched for.
 *
 * Renders scores provided by the walkscore api.
 *
 * Routed at /neighborhood/:neighborhood
 */
const Neighborhood = () => {
  const { neighborhood } = useParams();
  const navigate = useNavigate();
  const cityInfo = useSelector((st) => {
    for (let val of st.recentSearches) {
      if (val.city === neighborhood) return val;
    }
  });

  if (!cityInfo) navigate("/", { replace: true });

  return (
    <div className="Neighborhood text-center m-3">
      <h1>{cityInfo.city}</h1>
      <div className="row d-flex justify-content-center">
        <div className="col-lg-5 col-xl-4">
          {cityInfo.scores ? <Scores scores={cityInfo.scores} /> : ""}
          <a
            href={`https://www.zillow.com/homes/${cityInfo.city}_rb/`}
            target="_blank"
            className="btn btn-primary mb-3"
          >
            View available properties on Zillow
          </a>
        </div>

        <div className="col-lg-6 col-xl-7">
          <img
            src={cityInfo.image}
            alt={`${cityInfo.city}`}
            className="img-fluid img-thumbnail"
          />
        </div>
      </div>
    </div>
  );
};

export default Neighborhood;
