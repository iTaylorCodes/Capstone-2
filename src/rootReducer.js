import { v4 as uuidv4 } from "uuid";

const INITIAL_STATE = {
  accountWasDeleted: false,
  recentSearches: [],
};

const rootReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case "ACCOUNT_DELETED":
      return {
        ...state,
        accountWasDeleted: true,
      };

    case "RESET_ACCOUNT_DELETED":
      return {
        ...state,
        accountWasDeleted: false,
      };

    case "NEW_SEARCH":
      if (state.recentSearches.length >= 5) {
        state.recentSearches.shift();
      }

      return {
        ...state,
        recentSearches: [
          ...state.recentSearches,
          {
            city: action.searchTerm,
            image: action.image,
            scores: action.scores,
            id: uuidv4(),
          },
        ],
      };

    default:
      return state;
  }
};

export default rootReducer;
