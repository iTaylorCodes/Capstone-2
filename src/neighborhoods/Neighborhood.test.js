import Neighborhood from "./Neighborhood";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import * as redux from "react-redux";
import { createStore } from "redux";
import rootReducer from "../rootReducer";
import { MemoryRouter } from "react-router-dom";

const store = createStore(rootReducer);

const reduxSpy = jest.spyOn(redux, "useSelector");

beforeEach(() => {
  reduxSpy.mockReturnValue({
    city: "LA",
    image: "imageUrl",
    scores: {
      walkScore: { score: 10, description: "score" },
      bikeScore: { score: 10, description: "score" },
      transitScore: { score: 10, description: "score" },
    },
    id: expect.any(String),
  });
});

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <Neighborhood />
      </MemoryRouter>
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Neighborhood />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
