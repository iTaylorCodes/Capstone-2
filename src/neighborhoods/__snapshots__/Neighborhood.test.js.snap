// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Neighborhood text-center m-3"
  >
    <h1>
      LA
    </h1>
    <div
      class="row d-flex justify-content-center"
    >
      <div
        class="col-lg-5 col-xl-4"
      >
        <div
          class="Scores row p-1 d-flex justify-content-center"
        >
          <div
            class="col-4 col-md-3 col-lg-5"
          >
            <div
              class="Score m-3 rounded-circle p-2"
              style="width: 125px; height: 125px; background: rgb(93, 217, 93);"
            >
              <h2>
                10
              </h2>
              <p>
                score
              </p>
            </div>
          </div>
          <div
            class="col-4 col-md-3 col-lg-5"
          >
            <div
              class="Score m-3 rounded-circle p-2"
              style="width: 125px; height: 125px; background: rgb(93, 217, 93);"
            >
              <h2>
                10
              </h2>
              <p>
                score
              </p>
            </div>
          </div>
          <div
            class="col-4 col-md-3 col-lg-5"
          >
            <div
              class="Score m-3 rounded-circle p-2"
              style="width: 125px; height: 125px; background: rgb(93, 217, 93);"
            >
              <h2>
                10
              </h2>
              <p>
                score
              </p>
            </div>
          </div>
        </div>
        <a
          class="btn btn-primary mb-3"
          href="https://www.zillow.com/homes/LA_rb/"
          target="_blank"
        >
          View available properties on Zillow
        </a>
      </div>
      <div
        class="col-lg-6 col-xl-7"
      >
        <img
          alt="LA"
          class="img-fluid img-thumbnail"
          src="imageUrl"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
